packages/fast-components-react-base/src/toolbar/toolbar.tsx

```diff
@@ -0,0 +1,274 @@
+        initialFocusIndex: void 0,
+        orientation: void 0,
+        focusableRoles: void 0,
+        allowFocusOnDisabledItems: void 0,
+    };
+
+    private rootElement: React.RefObject<HTMLDivElement> = React.createRef<
+        HTMLDivElement
+    >();
+
+    constructor(props: ToolbarProps) {
+        super(props);
+
+        this.state = {
+            focusIndex: isNil(this.props.initialFocusIndex)
+                ? -1
+                : this.props.initialFocusIndex,
+        };
+    }
+
+    /**
+     * Renders the component
+     */
+    public render(): React.ReactElement<HTMLDivElement> {
+        return (
+            <div
+                {...this.unhandledProps()}
+                ref={this.rootElement}
+                role="toolbar"
+                className={this.generateClassNames()}
+                onKeyDown={this.handleMenuKeyDown}
+                onFocusCapture={this.handleItemFocus}
+            >
+                {this.renderChildren()}
+            </div>
+        );
+    }
+
+    public componentDidMount(): void {
+        if (this.state.focusIndex === -1) {
+            const children: Element[] = this.domChildren();
+            this.setState({
+                focusIndex: children.findIndex(this.isFocusableElement),
+            });
+        }
+        if (this.props.enableAutoFocus) {
+            this.focus();
+        }
+    }
+
+    /**
+     * Brings focus to the appropriate menu-item
+     */
+    public focus(): void {
+        this.setFocus(this.state.focusIndex === -1 ? 0 : this.state.focusIndex, 1);
+    }
+
+    /**
+     * Create class names
+     */
+    protected generateClassNames(): string {
+        const {
+            toolbar,
+            toolbar__horizontal,
+            toolbar__vertical,
+        }: ToolbarClassNameContract = this.props.managedClasses;
+        const isVertical: boolean = this.props.orientation === Orientation.vertical;
+
+        return super.generateClassNames(
+            classNames(
+                this.props.managedClasses.toolbar,
+                [toolbar__vertical, isVertical],
+                [toolbar__horizontal, !isVertical]
+            )
+        );
+    }
+
+    /**
+     * Render all child elements
+     */
+    private renderChildren(): React.ReactChild[] {
+        return React.Children.map(this.props.children, this.renderChild);
+    }
+
+    /**
+     * Render a single child
+     */
+    private renderChild = (
+        child: React.ReactElement,
+        index: number
+    ): React.ReactChild => {
+        return React.cloneElement(child, {
+            tabIndex: index === this.state.focusIndex ? 0 : -1,
+        });
+    };
+
+    /**
+     * Determines if a given element should be focusable by the menu
+     */
+    private isFocusableElement = (element: Element): element is HTMLElement => {
+        return (
+            element instanceof HTMLElement &&
+            this.props.focusableRoles.indexOf(element.getAttribute("role")) !== -1 &&
```

Comment:
How is it possible that there's no way to get the default role for an element? That seems like a miss in the spec to me.

Reply:
Elements like "div" would have not any default role.  But my next iteration should remove the roles based checks.

Reply:
we no longer depend on roles in next iteration